### ISSUE TITLE
Code generator does not produce indexes for quoted identifiers in PostgreSQL

### ISSUE THREAD

**Issue by mkurz** [jOOQ/jOOQ#11047]
Using jOOQ 3.14.4, OpenJdk 11 and Postgres 13
<hr>

Given:
```sql
create table some_table (
  "A" bigint not null
);
create index some_index on some_table ("A");
```
`some_index` is **not** added to`Indexes.java`

However, when unquoted (resulting in lower case), `some_index` **is** added to`Indexes.java`
```sql
create table some_table (
  A bigint not null
);
create index some_index on some_table (A);
```

I had a look at the code...
https://github.com/jOOQ/jOOQ/blob/222d2b49616741b00b5e1bef444cf05413596ede/jOOQ-meta/src/main/java/org/jooq/meta/postgres/PostgresDatabase.java#L185-L205
translates to following query:
```
# select alias_107226086.nspname, trel.relname, alias_76360422.relname, i.indisunique, array(select pg_get_indexdef(i.indexrelid, k + 1, true) from generate_subscripts(i.indkey, 1) as k order by k) as columns, i.indoption::int[] as asc_or_desc from (pg_catalog.pg_index as i join pg_catalog.pg_class as alias_76360422 on i.indexrelid = alias_76360422.oid) join (pg_catalog.pg_class as trel join pg_catalog.pg_namespace as alias_107226086 on trel.relnamespace = alias_107226086.oid) on trel.oid = i.indrelid where (alias_107226086.nspname in ('public') and (alias_107226086.nspname, alias_76360422.relname) not in (select alias_87501535.nspname, c.conname from (pg_catalog.pg_constraint as c join pg_catalog.pg_namespace as alias_87501535 on c.connamespace = alias_87501535.oid))) order by 1, 2, 3;
 nspname |              relname              |                   relname                   | indisunique |    columns     | asc_or_desc 
---------+-----------------------------------+---------------------------------------------+-------------+----------------+-------------
 public  | some_table                        | some_index                                  | f           | {"\"A\""}      | [0:0]={0}
```
See, the `columns` column contains `{"\"A\""}` (when unquoted, it's "just" `{a}`).

I am not 100% sure (because I didn't debug) but I guess the following `if` will now be entered, resulting in the index being ignored:
https://github.com/jOOQ/jOOQ/blob/222d2b49616741b00b5e1bef444cf05413596ede/jOOQ-meta/src/main/java/org/jooq/meta/postgres/PostgresDatabase.java#L224-L225
Looks like `table.getColumn("\"A\"")` will be `null`... 
A solution would be to strip the quotes or fix the query itself so the quotes won't be added in the first place.
(Also not sure if maybe `getColumn(...)` should be called with the `ignoreCase` param?)

**Comment by lukaseder** [comment]
Thanks for your report. I can reproduce this. In a way, it's similar as our current lack of support for function based indexes: #6310, which are also expressions reported by `pg_get_indexdef`, just like this is an expression.

Perhaps we could attempt using the parser to turn `"A"` into `A`. We'll have to use the parser anyway once we want to support expressions.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.15.0 and 3.14.5 (#11052)
